### PR TITLE
Feat/pipeline timeout

### PR DESCRIPTION
**Pull request by cheikhgwane**
Allow user to define timeout value after which the pipeline will be automatically stopped.

### CODE REVIEW

File: openhexa/sdk/pipelines/pipeline.py [hunk 1]
```diff
@@ -21,23 +21,28 @@
 from .utils import get_local_workspace_config
 
 logger = getLogger(__name__)
+PIPELINE_MAX_TIMEOUT = 43200
```
Comment (pvanliefland) [inline openhexa/sdk/pipelines/pipeline.py:24]: I don't think this should be hardcoded here. The max should be handled by the backend.

File: openhexa/sdk/pipelines/pipeline.py [hunk 2]
```diff
@@ -21,23 +21,28 @@
...
 
 
 class PipelineConfigError(Exception):
     pass
 
 
-def pipeline(code: str, *, name: str = None) -> typing.Callable[[typing.Callable[..., typing.Any]], "Pipeline"]:
+def pipeline(
+    code: str, *, name: str = None, timeout: int = PIPELINE_MAX_TIMEOUT
```
Comment (pvanliefland) [inline openhexa/sdk/pipelines/pipeline.py:32]: I would keep a default of `None` for the timeout and let the backend set the default value.

File: openhexa/sdk/pipelines/pipeline.py [hunk 3]
```diff
@@ -21,23 +21,28 @@
...
+) -> typing.Callable[[typing.Callable[..., typing.Any]], "Pipeline"]:
     if any(c not in string.ascii_lowercase + string.digits + "_-" for c in code):
         raise Exception("Pipeline name should contains only lower case letters, digits, '_' and '-'")
+    if timeout > PIPELINE_MAX_TIMEOUT:
```
Comment (pvanliefland) [inline openhexa/sdk/pipelines/pipeline.py:36]: Validation should be handled by the backend.

**Review by pvanliefland — CHANGES_REQUESTED** [review]
Hey @cheikhgwane , I think that if we want to specify timeouts through the SDK, we should keep the validation logic in the backend.

Also pinging @yannforget and @qgerome to get their opinion on this. Another option would be to specify timeouts through the web interface.

**Comment by cheikhgwane** [comment]
@pvanliefland Alright, will remove the validation on the sdk (already handled on the backend )

**Comment by qgerome** [comment]
> Hey @cheikhgwane , I think that if we want to specify timeouts through the SDK, we should keep the validation logic in the backend.
> 
> Also pinging @yannforget and @qgerome to get their opinion on this. Another option would be to specify timeouts through the web interface.

I agree with your comments. We had a discussion with Cheikh and the conclusion was to only implement it on the SDK & backend. The timeout can still be edited in the admin if needed but I would not let the user edit the timeout from the UI in the first version.

**Review by pvanliefland — APPROVED** [review]
LGTM!

**Pull request merged**